### DISCUSSION TITLE
Gatsby Deployment Issue in nginx

### DISCUSSION THREAD

**Discussion by bipin5** [gatsbyjs/gatsby#37651] [Help] [answered]
Hello, I have a gatsby project built which has a form that submits to a hubspot api. I have created a function in src/api folder as hubspot.js. So, when submitting the form, this api needs to be hit for the data to submit to hubspot. It is working perfectly in development but I get an error "POST http://localhost:3000/api/hubspot 404 (Not Found)" when in production using "serve public" and also in deployment using nginx (where it returns 405).
But, it is working fine when deployed to gatsby cloud.
Can someone help me figure what the issue is here? Thank you.

**Comment by ZeldOcarina** [comment, marked as answer]
Hi!
Gatsby functions don't work on a standard NGINX environment.
You need to either create your own serverless Lambda functions in AWS and hit those or create a standard Node/Express backend to serve that code.
I know well as I did the latter until Vercel finally begun supporting Gatsby a couple of weeks ago!

**Reply by bipin5** [reply]
Okay, I will do that too. Thanks for the help @ZeldOcarina.

**Reply by ZeldOcarina** [reply]
Happy to help! ♥️
